### PR TITLE
feat: embed default config file at compile time

### LINKED ISSUE #1771: Allow running without config specified
**Is your feature request related to a problem? Please describe.**
After we publish binary files along with release, it might be inconvenient to run qdrant with, as it always require config file to be present.

**Describe the solution you'd like**
Move all default values into the Rust source and allow running qdrant without config specified.
It should log the warning, at the start, that the default config is used.

If `RUN_MODE` env is specified, absence of configuration file should not be ignore

### PR DESCRIPTION
**Pull request by paulotten**
Closes #1771.

This is an alternative to #1773.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally using `cargo fmt` command prior to submission?
3. [x] Have you checked your code using `cargo clippy` command?

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -1,3 +1,6 @@
+debug: false
+log_level: INFO
```
Comment (paulotten) [inline config/config.yaml:2]: Copied over from production.yaml

As per https://github.com/qdrant/qdrant/issues/1771#issuecomment-1518735959 we want to default to a production like configuration.

**Comment by paulotten** [comment]
> It should log the warning, at the start, that the default config is used.

> If RUN_MODE env is specified, absence of configuration file should not be ignore

@generall Do you want me to copy over that code from #1773 to here?

**Comment by timvisee** [comment]
> > It should log the warning, at the start, that the default config is used.
> 
> > If RUN_MODE env is specified, absence of configuration file should not be ignore
> 
> Do you want me to copy over that code from #1773 to here?

Yes. Additionally we now have to check whether any of the configuration files exists. If not, show the warning:

```rust
log::warn!("Configuration files not found. Using default configuration.");
```

File: config/config.yaml (hunk 1 above)
Comment (timvisee) [reply config/config.yaml:2]: Yes! In development mode it would simply pick up the `config/development.yaml` at runtime, fantastic.

**Review by timvisee — CHANGES_REQUESTED** [review]
Looks good. Simple yet effective. Thanks!

The only change I request is testing the embedded configuration in a unit test as discussed.

**Review by generall — APPROVED** [review]

File: src/settings.rs [hunk 2]
```diff
@@ -231,12 +261,26 @@ mod tests {
...
 
         // Validate
         config.validate().unwrap();
+        assert!(config.found_config_files)
+    }
+
+    #[test]
+    fn test_no_config_files() {
+        let non_existing_config_path = "config/non_existing_config".to_string();
+        env::remove_var("RUN_MODE");
```
Comment (paulotten) [inline src/settings.rs:277]: @generall this combined with the `env::set_var` in the above test cause a race condition.

Comment (timvisee) [reply src/settings.rs:292]: Correct. Environment variables are a shared resources, and tests can be run concurrently.

I've added a test that ensures the embedded configuration can be loaded on it's own. I think that's the best we can do without cluttering the `Settings` interface.

**Review by timvisee — APPROVED** [review]

File: src/settings.rs (hunk 2 above)
Comment (agourlay) [reply src/settings.rs:292]: happened several times so I have created https://github.com/qdrant/qdrant/issues/1803

**Pull request merged**